src/bundle/Controller/LocationController.php

```diff
@@ -377,6 +396,48 @@ public function trashContainerAction(Request $request): Response
         return $this->redirectToLocation($form->getData()->getLocation());
     }
 
+    /**
+     * @param \EzSystems\EzPlatformAdminUi\Form\Data\Location\LocationTrashData|\EzSystems\EzPlatformAdminUi\Form\Data\Location\LocationTrashContainerData $data
+     *
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    private function handleTrashLocation(LocationTrashData $data): RedirectResponse
+    {
+        $location = $data->getLocation();
+        $parentLocation = $this->locationService->loadLocation($location->parentLocationId);
+        $trashOptions = $data->getTrashOptions();
+
+        $this->repository->beginTransaction();
+        try {
+            if (isset($trashOptions[HasUniqueAssetRelation::TRASH_ASSETS])
+                && HasUniqueAssetRelation::RADIO_SELECT_TRASH_WITH_ASSETS === $trashOptions[HasUniqueAssetRelation::TRASH_ASSETS]
+            ) {
+                $this->trashRelatedAsset($location->getContentInfo());
+            }
+            $this->trashService->trash($location);
+            $this->repository->commit();
+        } catch (\Exception $exception) {
+            $this->repository->rollback();
+            throw $exception;
+        }
+
+        $this->notificationHandler->success(
+            $this->translator->trans(
+                /** @Desc("Location '%name%' moved to Trash.") */
+                'location.trash.success',
+                ['%name%' => $location->getContentInfo()->name],
+                'location'
+            )
+        );
+
+        return new RedirectResponse($this->generateUrl('_ezpublishLocation', [
+            'locationId' => $parentLocation->id,
+        ]));
```

Comment:
Might be replaced by `return $this->redirectToLocation($parentLocation);` if controller extends `\EzSystems\EzPlatformAdminUiBundle\Controller\Controller`.